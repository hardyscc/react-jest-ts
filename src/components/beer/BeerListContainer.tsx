import React from "react";
import BeerList from "./BeerList";
import InputArea from "./InputArea";

interface IProps {
  title?: string;
}

interface IState {
  beers: string[];
}

class BeerListContainer extends React.Component<IProps, IState> {
  static defaultProps: Partial<IProps> = {
    title: "Beer List"
  };

  state: IState = {
    beers: []
  };

  render() {
    return (
      <React.Fragment>
        <InputArea onSubmit={this.addItem} />
        <BeerList items={this.state.beers} />
      </React.Fragment>
    );
  }

  addItem = (name: string) => {
    this.setState({
      beers: [...this.state.beers, name]
    });
  };
}

export default BeerListContainer;
