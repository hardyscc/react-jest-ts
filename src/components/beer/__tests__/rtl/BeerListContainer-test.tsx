import React from "react";
import { fireEvent, render, wait } from "react-testing-library";
import BeerListContainer from "../../BeerListContainer";

describe("react-testing-library - BeerListContainer", () => {
  it("add beer", async () => {
    const { getByText, getByLabelText, queryByText, container } = render(
      <BeerListContainer />
    );

    const inputText = getByLabelText(/name/i);
    const addButton = getByText(/add/i);

    fireEvent.change(inputText, {
      target: { name: "name", value: "Octoberfest" }
    });
    fireEvent.click(addButton);

    await wait(() => expect(queryByText("Octoberfest")).not.toBeNull());
    expect(container).toMatchSnapshot();

    fireEvent.change(inputText, {
      target: { name: "name", value: "Killingbear" }
    });
    fireEvent.click(addButton);

    await wait(() => expect(queryByText("Killingbear")).not.toBeNull());
    expect(container).toMatchSnapshot();
  });
});
