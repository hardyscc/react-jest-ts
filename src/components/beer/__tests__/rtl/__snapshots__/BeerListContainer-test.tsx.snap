// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-testing-library - BeerListContainer add beer 1`] = `
<ul>
  <li>
    Octoberfest
  </li>
</ul>
`;

exports[`react-testing-library - BeerListContainer add beer 2`] = `
<ul>
  <li>
    Octoberfest
  </li>
  <li>
    Killingbear
  </li>
</ul>
`;
